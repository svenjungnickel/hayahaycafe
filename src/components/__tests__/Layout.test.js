import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../Layout';

jest.mock('../SEO');
jest.mock('../NavBar');
jest.mock('../Footer/Footer');

describe('Layout', () => {
    it('renders without required children throws prop type error', () => {
        const renderLayout = () => {
            const meta = {};
            const currentPage = 'currentPage';

            const component = <Layout meta={meta} currentPage={currentPage} />;
            renderer.create(component);
        };

        expect(renderLayout).toThrowError('Warning: Failed prop type');
    });

    it('renders without required meta throws prop type error', () => {
        const renderLayout = () => {
            const children = <div>children</div>;
            const currentPage = 'currentPage';

            const component = <Layout currentPage={currentPage}>{children}</Layout>;
            renderer.create(component);
        };

        expect(renderLayout).toThrowError('Warning: Failed prop type');
    });

    it('renders without required currentPage throws prop type error', () => {
        const renderLayout = () => {
            const children = <div>children</div>;
            const meta = {};

            const component = <Layout meta={meta}>{children}</Layout>;
            renderer.create(component);
        };

        expect(renderLayout).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid children throws prop type error', () => {
        const renderLayout = () => {
            const children = null;
            const meta = {};
            const currentPage = 'currentPage';

            const component = (
                <Layout meta={meta} currentPage={currentPage}>
                    {children}
                </Layout>
            );
            renderer.create(component);
        };

        expect(renderLayout).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid meta throws prop type error', () => {
        const renderLayout = () => {
            const children = <div>children</div>;
            const meta = false;
            const currentPage = 'currentPage';

            const component = (
                <Layout meta={meta} currentPage={currentPage}>
                    {children}
                </Layout>
            );
            renderer.create(component);
        };

        expect(renderLayout).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid currentPage throws prop type error', () => {
        const renderLayout = () => {
            const children = <div>children</div>;
            const meta = {};
            const currentPage = 123;

            const component = (
                <Layout meta={meta} currentPage={currentPage}>
                    {children}
                </Layout>
            );
            renderer.create(component);
        };

        expect(renderLayout).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const children = <div>children</div>;
        const meta = {};
        const currentPage = 'currentPage';

        const component = (
            <Layout meta={meta} currentPage={currentPage}>
                {children}
            </Layout>
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
