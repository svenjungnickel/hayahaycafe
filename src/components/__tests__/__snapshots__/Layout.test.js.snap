// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<main>
  <head>
    <title>
      SEO
    </title>
  </head>
  <nav />
  <div>
    children
  </div>
  <footer />
</main>
`;
